tests/unit_tests/test_surface.py

```diff
@@ -329,14 +330,28 @@ def test_quadric():
 
 
 def test_cylinder_from_points():
-    # Generate 45-degree rotated cylinder in x-y plane with radius 1
-    p1 = (0, 0, 0)
-    p2 = (1, 1, 0)
-    s = openmc.model.cylinder_from_points(p1, p2, 1)
-
-    # Points p1 and p2 need to be inside cylinder
-    assert p1 in -s
-    assert p2 in -s
-    assert (-1, 1, 0) in +s
-    assert (1, -1, 0) in +s
-    assert (0, 0, 1.5) in +s
+    for _ in range(10):
+        # Generate cylinder in random direction
+        p1 = np.array([random(), random(), random()])
+        p2 = np.array([random(), random(), random()])
+        r = random()
```

Comment:
In some rare cases the test might fail because of floating-point arithmetic, e.g. if `r` is very small.

Reply:
Did you actually run into such a case?

Reply:
Not running it as is, but playing around with the values it may fail if `r` is less than ~1e-6. So it's not very likely we'd run into that case.

Reply:
Ok, I'll just have it do a random radius that is greater than 1